ethexe/consensus/src/validator/subordinate.rs

```diff
@@ -0,0 +1,399 @@
+/// It waits for the producer block, the waits for the block computing
+/// and then switches to [`Participant`] state.
+#[derive(Debug, Display)]
+#[display("SUBORDINATE in {:?}", self.state)]
+pub struct Subordinate {
+    ctx: ValidatorContext,
+    producer: Address,
+    block: SimpleBlockData,
+    is_validator: bool,
+    state: State,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+enum State {
+    WaitingForProducerBlock,
+    WaitingProducerBlockComputed {
+        // TODO +_+_+: change this to producer-block digest when off-chain transactions added
+        block_hash: H256,
+    },
+}
+
+impl StateHandler for Subordinate {
+    fn into_dyn(self: Box<Self>) -> Box<dyn StateHandler> {
+        self
+    }
+
+    fn context(&self) -> &ValidatorContext {
+        &self.ctx
+    }
+
+    fn context_mut(&mut self) -> &mut super::ValidatorContext {
+        &mut self.ctx
+    }
+
+    fn into_context(self: Box<Self>) -> ValidatorContext {
+        self.ctx
+    }
+
+    fn process_block_from_producer(
+        mut self: Box<Self>,
+        block: SignedData<ProducerBlock>,
+    ) -> Result<Box<dyn StateHandler>> {
+        if self.state == State::WaitingForProducerBlock
+            && block.verify_address(self.producer).is_ok()
+            && block.data().block_hash == self.block.hash
+        {
+            let pb = block.into_parts().0;
+            let block_hash = pb.block_hash;
+
+            self.output(ConsensusEvent::ComputeProducerBlock(pb));
+
+            self.state = State::WaitingProducerBlockComputed { block_hash };
+
+            Ok(self)
+        } else {
+            DefaultProcessing::block_from_producer(self, block)
+        }
+    }
+
+    fn process_validation_request(
+        mut self: Box<Self>,
+        request: SignedData<crate::BatchCommitmentValidationRequest>,
+    ) -> Result<Box<dyn StateHandler>> {
+        if request.verify_address(self.producer).is_ok() {
+            log::trace!("Receive validation request from producer: {request:?}, saved for later.");
+            self.ctx.pending(request);
+
+            Ok(self)
+        } else {
+            DefaultProcessing::validation_request(self, request)
+        }
+    }
+
+    fn process_computed_block(
+        self: Box<Self>,
+        computed_block: H256,
+    ) -> Result<Box<dyn StateHandler>> {
+        match &self.state {
+            _ if computed_block == self.block.header.parent_hash => {
+                // Earlier we sent a task for parent block computation.
+                // Continue to wait for block from producer.
+                Ok(self)
+            }
+            State::WaitingProducerBlockComputed { block_hash } if computed_block == *block_hash => {
+                if self.is_validator {
+                    Participant::create(self.ctx, self.block, self.producer)
+                } else {
+                    Initial::create(self.ctx)
+                }
+            }
+            _ => DefaultProcessing::computed_block(self, computed_block),
```

Comment:
Please, add comment that we switch to default processing because of possible reorganization

Reply:
Sorry, could you explain what should be in the comment? Because line 
```rust
_ => DefaultProcessing::...
```
exactly tells us, that we switch here to default proccessing

Reply:
it tells what we do, but not why